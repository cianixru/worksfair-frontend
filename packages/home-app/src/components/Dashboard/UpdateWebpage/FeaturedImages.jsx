import React, { Component } from 'react';
import Dropzone from 'react-dropzone';
import axios from 'axios';
import PropTypes from 'prop-types';
import { Image, Transformation } from 'cloudinary-react';

import {
  cloudinaryImageUploader,
  cloudinaryImageRemover,
} from '../../utils/helpers';

class FeaturedImages extends Component {
  state={
    existingImages: (this.props.webpage
      && this.props.webpage.featured_images) || [],
    selectedImages: [],
    uploadedImages: (this.props.webpage
      && this.props.webpage.featured_images) || [],
    rawFiles: {},
    setBackground: (this.props.webpage
    && this.props.webpage.set_background) || false,
    updated: false,
  }

  /**
   * @description Handles the selection of the images
   *
   * @param { Array } images
   *
   * @returns { Promise } axios API call
   */

  handleImagesSelection = (files) => {
    const rawFiles = {};
    const localFileURLs = files.map((file) => {
      const localFileURL = URL.createObjectURL(file);
      rawFiles[localFileURL] = file;
      return localFileURL;
    });

    // console.log(files, localFileURLs, rawFiles);
    const { selectedImages } = this.state;
    this.setState({
      ...selectedImages,
      selectedImages: localFileURLs,
      rawFiles,
    });
  }

  /**
   * @description Handles the response from cloudinary API call
   *
   * @param { array } images
   */
  handleCloudinaryResponse = async (images) => {
    const { uploadedImages } = this.state;
    const uploads = images.map((image) => {
      return cloudinaryImageUploader(image)
        .then((imageData) => {
          uploadedImages.push(imageData.data.secure_url);
          this.setState({
            uploadedImages,
          });
        });
    });
    return uploads;
  }

  /**
   * @description Handles the removal of uploaded image from preview
   *
   * @param { string } imageURL
   *
   * @returns { Promise } axios API call
   */
  handleRemoveImage = imageURL => () => {
    const { selectedImages } = this.state;

    // find and remove the image from the selectedImages array
    selectedImages.splice(selectedImages.indexOf(imageURL), 1);

    this.setState({
      selectedImages,
    });
  }

  /**
   * @description Handles the removal of uploaded image from cloud
   * and from the preview
   *
   * @param { string } imageCloudURL
   *
   * @returns { Promise } axios API call
   */
  handleRemoveImagefromCloud = imageCloudURL => () => {
    this.props.isLoading();
    try {
      cloudinaryImageRemover(imageCloudURL)
        .then((response) => {
          if (response.data.result === 'ok') {
            const { uploadedImages } = this.state;
            // remove the image from the state
            uploadedImages.splice(uploadedImages.indexOf(imageCloudURL), 1);
            this.props.onSubmit({
              featured_images: uploadedImages,
            })
              .then(() => {
                this.setState({
                  existingImages: uploadedImages,
                  uploadedImages,
                });
              });
          }
        });
    } catch (error) {
      console.log(error);
    } finally {
      this.props.isComplete();
    }
  }

  handleCheckboxChange = (event) => {
    event.preventDefault();
    this.setState({
      setBackground: event.target.checked,
      updated: true,
    })
  }

  /**
   * @description sends the cloudinary urls to the DB
   */
  handleSubmit = async (event) => {
    event.preventDefault();
    const { uploadedImages, selectedImages, rawFiles, setBackground } = this.state;
    const { isComplete, isLoading } = this.props;

    isLoading();
    const imagesToUpload = [];
    selectedImages.map((imageBlob) => {
      return imagesToUpload.push(rawFiles[imageBlob]);
    });

    const imagesData = await this.handleCloudinaryResponse(imagesToUpload);
    axios.all(imagesData).then(() => {
      const images = {
        featured_images: uploadedImages,
        set_background: setBackground,
      };
      this.props.onSubmit(images);
      this.setState({
        selectedImages: [],
      });
      isComplete();
      console.log('Images have all being uploaded to cloudinary');
    });
  }

  render() {
    const {
      selectedImages,
      existingImages,
      setBackground,
      updated,
    } = this.state;
    const hasImagesOrUpdated =  (updated || selectedImages.length > 0);

    return (
      <Dropzone
        onDrop={this.handleImagesSelection}
        multiple
        accept="image/*"
      >
        {({ getRootProps, getInputProps }) => {
          // The list of existing pictures from the DB
          const existingPictureList = existingImages.map((imageCloudURL, index) => (
            <div
              key={`${imageCloudURL}${index}`}
              className="column is-one-quarter"
            >
              <div className="image-preview">
                <Image
                  cloudName="worksfair"
                  publicId={imageCloudURL}
                  alt={imageCloudURL}
                  type="fetch">
                  <Transformation width="200" height="200" fetchFormat="auto" crop="fit" />
                </Image>
                <div className="image-preview-overlay">
                  <button
                    className="delete is-large is-pulled-right"
                    onClick={this.handleRemoveImagefromCloud(imageCloudURL)}
                    data-testid="delete-image"
                  />
                </div>
              </div>
            </div>
          ));

          // The list of newly selected images
          const pictureList = selectedImages.map((localObjectURL, index) => (
            <div
              key={`${localObjectURL}${index}`}
              className="column is-one-quarter"
            >
              <div className="image-preview">
                <img
                  src={localObjectURL}
                  alt={localObjectURL}
                />
                <div className="image-preview-overlay">
                  <button
                    className="delete is-large is-pulled-right"
                    onClick={this.handleRemoveImage(localObjectURL)}
                    data-testid="delete-image"
                  />
                </div>
              </div>
            </div>
          ));

          return (
            <div>
              <div className="drop-area">
                <div
                  {...getRootProps({ className: 'dropzone' })}
                >
                  <input {...getInputProps()} />
                  <p>Try dropping some pictures here, or click to select pictures to upload.</p>
                </div>
                <div className="columns is-multiline ">
                  {existingPictureList}
                  {pictureList}
                </div>
              </div>
                <div className="control">
                  <label className="checkbox">
                    <input
                      type="checkbox"
                      name="set-background"
                      checked={setBackground}
                      onChange={this.handleCheckboxChange}
                    />
                    Set the first picture as your website background image
                  </label>
                </div>
                <div className="control">
                  <button
                    type="submit"
                    disabled={!hasImagesOrUpdated}
                    className="button is-link is-medium is-rounded"
                    data-testid="save-featured-images"
                    onClick={this.handleSubmit}
                  >
                    Save & Continue
                  </button>
                </div>
            </div>
          );
        }}
      </Dropzone>
    );
  }
}

FeaturedImages.propTypes = {
  onSubmit: PropTypes.func,
  webpage: PropTypes.object,
  isLoading: PropTypes.func,
  isComplete: PropTypes.func,
};

export default FeaturedImages;
