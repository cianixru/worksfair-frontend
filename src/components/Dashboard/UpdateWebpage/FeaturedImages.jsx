import React, { Component } from 'react';
import Dropzone from 'react-dropzone';
import axios from 'axios';
import sha1 from 'sha1';
import PropTypes from 'prop-types';

import {
  CloudinaryImageUploader,
  extractCloudID,
} from '../../utils/helpers';

const {
  REACT_APP_CLOUDINARY_API_KEY,
  REACT_APP_CLOUDINARY_PRESET_NAME,
  REACT_APP_CLOUDINARY_DELETE_URL,
  REACT_APP_CLOUDINARY_AUTHORIZATION,
} = process.env;
class FeaturedImages extends Component {
  state={
    existingImages: (this.props.webpage
      && this.props.webpage.featured_images) || [],
    selectedImages: [],
    uploadedImages: (this.props.webpage
      && this.props.webpage.featured_images) || [],
    rawFiles: {},
  }

  /**
   * @description Handles the selection of the images
   *
   * @param { Array } images
   *
   * @returns { Promise } axios API call
   */

  handleImagesSelection = (files) => {
    const rawFiles = {};
    const localFileURLs = files.map((file) => {
      const localFileURL = URL.createObjectURL(file);
      rawFiles[localFileURL] = file;
      return localFileURL;
    });

    // console.log(files, localFileURLs, rawFiles);
    const { selectedImages } = this.state;
    this.setState({
      ...selectedImages,
      selectedImages: localFileURLs,
      rawFiles,
    });
  }

  /**
   * @description Handles the response from cloudinary API call
   *
   * @param { array } images
   */
  handleCloudinaryResponse = async (images) => {
    const { uploadedImages } = this.state;
    const uploads = images.map((image) => {
      return CloudinaryImageUploader(image)
        .then((imageData) => {
          uploadedImages.push(imageData.data.secure_url);
          this.setState({
            uploadedImages,
          });
        });
    });
    return uploads;
  }

  /**
   * @description Handles the removal of uploaded image from preview
   *
   * @param { string } imageURL
   *
   * @returns { Promise } axios API call
   */
  handleRemoveImage = imageURL => () => {
    const { selectedImages } = this.state;

    // find and remove the image from the selectedImages array
    selectedImages.splice(selectedImages.indexOf(imageURL), 1);

    this.setState({
      selectedImages,
    });
  }

  /**
   * @description Handles the removal of uploaded image from cloud
   * and from the preview
   *
   * @param { string } imageCloudURL
   *
   * @returns { Promise } axios API call
   */
  handleRemoveImagefromCloud = imageCloudURL => () => {
    this.props.isLoading();
    try {
      const imagePublicID = extractCloudID(imageCloudURL);
      // our formdata
      const timestamp = Date.now();
      const signature = sha1(`public_id=${imagePublicID}&timestamp=${
        timestamp.toString()}${REACT_APP_CLOUDINARY_AUTHORIZATION}`);

      const formData = new FormData();
      formData.append('public_id', imagePublicID);
      formData.append(
        'api_key',
        REACT_APP_CLOUDINARY_API_KEY
      );
      formData.append(
        'upload_preset',
        REACT_APP_CLOUDINARY_PRESET_NAME
      );
      formData.append('timestamp', timestamp);
      formData.append('signature', signature);

      return axios({
        method: 'post',
        url: REACT_APP_CLOUDINARY_DELETE_URL,
        data: formData,
        config: { headers: { 'Content-Type': 'multipart/form-data' } }
      })
        .then((response) => {
          if (response.data.result === 'ok') {
            const { uploadedImages } = this.state;
            // remove the image from the state
            uploadedImages.splice(uploadedImages.indexOf(imageCloudURL), 1);
            this.props.onSubmit({
              featured_images: uploadedImages,
            })
              .then(() => {
                this.setState({
                  existingImages: uploadedImages,
                  uploadedImages,
                });
              });
          }
        });
    } catch (error) {
      console.log(error);
    } finally {
      this.props.isComplete();
    }
  }

  /**
   * @description sends the cloudinary urls to the DB
   */
  handleSubmit = async () => {
    const { uploadedImages, selectedImages, rawFiles } = this.state;
    const { isComplete, isLoading } = this.props;

    isLoading();
    const imagesToUpload = [];
    selectedImages.map((imageBlob) => {
      return imagesToUpload.push(rawFiles[imageBlob]);
    });

    const imagesData = await this.handleCloudinaryResponse(imagesToUpload);
    axios.all(imagesData).then(() => {
      const images = {
        featured_images: uploadedImages,
      };
      this.props.onSubmit(images);
      this.setState({
        selectedImages: [],
      });
      isComplete();
      console.log('Images have all being uploaded to cloudinary');
    });
  }

  render() {
    const { selectedImages, existingImages } = this.state;
    const hasNoImages = selectedImages.length <= 0;

    return (
      <Dropzone
        onDrop={this.handleImagesSelection}
        multiple
        accept="image/*"
      >
        {({ getRootProps, getInputProps }) => {
          // The list of existing pictures from the DB
          const existingPictureList = existingImages.map((imageCloudURL, index) => (
            <li
              key={`${imageCloudURL}${index}`}
              className="notification column is-one-quarter"
            >
              <button
                className="delete"
                onClick={this.handleRemoveImagefromCloud(imageCloudURL)}
                data-testid="delete-image"
              />
              <figure className="image is-180x180">
                <img
                  src={imageCloudURL}
                  alt={imageCloudURL}
                />
              </figure>
            </li>
          ));

          // The list of newly selected images
          const pictureList = selectedImages.map((localObjectURL, index) => (
            <li
              key={`${localObjectURL}${index}`}
              className="notification column is-one-quarter"
            >
              <button
                className="delete"
                onClick={this.handleRemoveImage(localObjectURL)}
                data-testid="delete-image"
              />
              <figure className="image is-180x180">
                <img
                  src={localObjectURL}
                  alt={localObjectURL}
                />
              </figure>
            </li>
          ));

          return (
            <div>
              <div className="drop-area">
                <div
                  {...getRootProps({ className: 'dropzone' })}
                >
                  <input {...getInputProps()} />
                  <p>Try dropping some pictures here, or click to select pictures to upload.</p>
                </div>
                <ul className="columns is-multiline ">
                  {existingPictureList}
                  {pictureList}
                </ul>
              </div>
              <div className="control">
                <button
                  type="submit"
                  disabled={hasNoImages}
                  className="button is-danger is-medium is-rounded"
                  data-testid="save-featured-images"
                  onClick={this.handleSubmit}
                >
                  Save & Continue
                </button>
              </div>
            </div>
          );
        }}
      </Dropzone>
    );
  }
}

FeaturedImages.propTypes = {
  onSubmit: PropTypes.func,
  webpage: PropTypes.object,
  isLoading: PropTypes.func,
  isComplete: PropTypes.func,
};

export default FeaturedImages;
