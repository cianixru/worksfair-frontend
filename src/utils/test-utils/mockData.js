// eslint-disable-next-line import/prefer-default-export
export const user = {
  username: 'Theookafor2',
  first_name: 'eod',
  last_name: 'okafor',
  email: 'theodorehope@andela.com',
  facebook: null,
  instagram: null,
  twitter: null,
  headline: null,
  current_location: null,
  role: 'admin',
  confirmed_account: false,
  verified_account: false,
  reset_token: null,
  image_url: null,
  created_at: '2019-03-10T08:48:42.942532Z',
  modified_at: '2019-03-10T08:48:42.942544Z',
  token: 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9',
};

export const webpage = {
  owner: {
    id: 1,
    username: 'theowa',
    first_name: 'theo',
    last_name: 'IO',
    email: 'theo.iowa@yahoo.com',
    facebook: null,
    instagram: null,
    twitter: null,
    headline: null,
    current_location: null,
    role: 'admin',
    confirmed_account: false,
    verified_account: false,
    reset_token: null,
    image_url: 'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png',
    created_at: '2019-04-25T09:07:08.209916Z',
    modified_at: '2019-04-25T09:07:08.209930Z'
  },
  title: 'Happiness',
  sub_domain_name: 'happiness',
  description: 'Happiness truly lives here. We just happy',
  keywords: 'passion, software, engineering, technology, worksfair',
  country: 'Nigeria',
  state: 'Lagos',
  city: 'Okota',
  area: 'Omagba Phase 1, Phase 2, Abatete',
  address: 'No 7 Obinna Uzor Street, Canal Estate',
  phone: null,
  email: null,
  website: null,
  featured_images: [
    'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png',
    'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png'
  ],
  is_active: true,
  logo: null,
  colour: 'is-dark',
  created_at: '2019-04-28T18:39:25.174272Z',
  updated_at: '2019-05-01T12:18:49.669633Z'
};
