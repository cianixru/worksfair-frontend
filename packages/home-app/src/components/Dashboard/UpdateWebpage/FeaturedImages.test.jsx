import React from 'react';
import { Provider } from 'react-redux';
import { render, cleanup, fireEvent } from 'react-testing-library';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';

import FeaturedImages from './FeaturedImages';
import { webpage } from '../../../utils/test-utils/mockData';

afterEach(cleanup);

const middlewares = [thunk];
const store = configureMockStore(middlewares);
const mockStore = store({
  auth: { currentUser: {} },
  webpage: { newWebpage: {} }
});

describe('FeaturedImages.jsx', () => {
  const onSubmit = jest.fn();
  const user = {};
  const validationErrors = {};
  const handleErrorReset = jest.fn();
  const actions = {
    isComplete: jest.fn(),
    isLoading: jest.fn(),
  };

  test('should submit the create webpage form data', () => {
    const { getByTestId } = render(
      <Provider store={mockStore}>
        <Router>
          <FeaturedImages
            onSubmit={onSubmit}
            user={user}
            validationErrors={validationErrors}
            handleErrorReset={handleErrorReset}
            webpage={webpage}
            isLoading={actions.isLoading}
            isComplete={actions.isComplete}
          />
        </Router>
      </Provider>
    );

    const saveImagesBtn = getByTestId('save-featured-images');
    saveImagesBtn.onclick = onSubmit;
    fireEvent.click(saveImagesBtn);

    expect(onSubmit).toHaveBeenCalledTimes(1);
    // expect(actions.isLoading).toHaveBeenCalledTimes(1);
  });

  test('that component renders correctly', async () => {
    const component = render(
      <Provider store={mockStore}>
        <Router>
          <FeaturedImages
            onSubmit={onSubmit}
            user={user}
            validationErrors={validationErrors}
            handleErrorReset={handleErrorReset}
            webpage={webpage}
            isLoading={actions.isLoading}
            isComplete={actions.isComplete}
          />
        </Router>
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
