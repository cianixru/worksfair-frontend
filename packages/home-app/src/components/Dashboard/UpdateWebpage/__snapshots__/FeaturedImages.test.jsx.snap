// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedImages.jsx that component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="drop-area"
        >
          <div
            class="dropzone"
            tabindex="0"
          >
            <input
              accept="image/*"
              autocomplete="off"
              multiple=""
              style="display: none;"
              tabindex="-1"
              type="file"
            />
            <p>
              Try dropping some pictures here, or click to select pictures to upload.
            </p>
          </div>
          <div
            class="columns is-multiline "
          >
            <div
              class="column is-one-quarter"
            >
              <div
                class="image-preview"
              >
                <img
                  alt="https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
                  src="http://res.cloudinary.com/worksfair/image/fetch/c_fit,f_auto,h_200,w_200/https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
                />
                <div
                  class="image-preview-overlay"
                >
                  <button
                    class="delete is-large is-pulled-right"
                    data-testid="delete-image"
                  />
                </div>
              </div>
            </div>
            <div
              class="column is-one-quarter"
            >
              <div
                class="image-preview"
              >
                <img
                  alt="https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
                  src="http://res.cloudinary.com/worksfair/image/fetch/c_fit,f_auto,h_200,w_200/https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
                />
                <div
                  class="image-preview-overlay"
                >
                  <button
                    class="delete is-large is-pulled-right"
                    data-testid="delete-image"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="control"
        >
          <label
            class="checkbox"
          >
            <input
              name="set-background"
              type="checkbox"
            />
            Set the first picture as your website background image
          </label>
        </div>
        <div
          class="control"
        >
          <button
            class="button is-link is-medium is-rounded"
            data-testid="save-featured-images"
            disabled=""
            type="submit"
          >
            Save & Continue
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="drop-area"
      >
        <div
          class="dropzone"
          tabindex="0"
        >
          <input
            accept="image/*"
            autocomplete="off"
            multiple=""
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <p>
            Try dropping some pictures here, or click to select pictures to upload.
          </p>
        </div>
        <div
          class="columns is-multiline "
        >
          <div
            class="column is-one-quarter"
          >
            <div
              class="image-preview"
            >
              <img
                alt="https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
                src="http://res.cloudinary.com/worksfair/image/fetch/c_fit,f_auto,h_200,w_200/https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
              />
              <div
                class="image-preview-overlay"
              >
                <button
                  class="delete is-large is-pulled-right"
                  data-testid="delete-image"
                />
              </div>
            </div>
          </div>
          <div
            class="column is-one-quarter"
          >
            <div
              class="image-preview"
            >
              <img
                alt="https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
                src="http://res.cloudinary.com/worksfair/image/fetch/c_fit,f_auto,h_200,w_200/https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
              />
              <div
                class="image-preview-overlay"
              >
                <button
                  class="delete is-large is-pulled-right"
                  data-testid="delete-image"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="control"
      >
        <label
          class="checkbox"
        >
          <input
            name="set-background"
            type="checkbox"
          />
          Set the first picture as your website background image
        </label>
      </div>
      <div
        class="control"
      >
        <button
          class="button is-link is-medium is-rounded"
          data-testid="save-featured-images"
          disabled=""
          type="submit"
        >
          Save & Continue
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
